### ISSUE TITLE
[en][item][spell][mage] Dragonwrath legendary staff and pyroblast  and Critical Mass

### ISSUE THREAD

**Issue by stanq87** [Atlantiss/BugTracker#8848]
**Description**: 

http://wow.gamepedia.com/Dragonwrath,_Tarecgosa%27s_Rest
http://cata-shoot.tauri.hu/?spell=22959
http://wow.gamepedia.com/Critical_Mass

**Current behaviour**: the duplicate pyroblast does not apply Critical Mass

**Expected behaviour**: it should apply Critical Mass

"Duplicated spells behave as if you did cast them. They are subject to the same talents, buffs, debuffs and other conditions."

**Steps to reproduce the problem**:

1. go spam pyroblasts
2. wait for a proc from staff
3. watch if it will refresh Critical Mass

**Revision**: 5294

**Comment by Batranelu** [comment]
This isn't such a big problem , your complain here is pyroblast proc from staff doesn't refresh 1 second of Critical Mass buff cause you already have refreshed Critical mass when you applied pyroblast so if staff would of apply too the buff would refresh in 1 sec after the apply so you lose 1 second from Critical Mass , not big deal but still a bug.

**Comment by stanq87** [comment]
i know its low priority, and not such a big deal, but a bug is a bug
its  independent but its  subject to talents  and Critical Mass is one,

**Comment by sigmundfreud2016** [comment]
@Batranelu you are absolutely correct, I can imagine only 1 situation where it changes something:
DK casts AMS just before getting bursted by FM, last pyroblast destroys his AMS and triggers duplicate pyroblast which hits now unprotected DK.

**Comment by Illusie** [comment]
Well the Mass Critical is bugged when you have 2 Fire mages in your group
or Warlock they keep remove/re-apply the Critical Mass everytime..

Atleast when I play my Firemage I feel like -2-3k dps If Warlock or another Fmage in my group.

**Comment by sigmundfreud2016** [comment]
@Illusie this is NOT bugged by any means. Critical mass and similar effects from different casters shouldn't stack, so they override each other by dispelling the previous effect applied, it's normal. I don't understand, why does it bother you: all such debuffs have exactly the same effect and work for any attacker. 
Try to play frost in PVE, you will be surprised how every single class (except healers) breaks your roots.

**Comment by Illusie** [comment]
@sigmundfreud2016 Yes the issue is that it always get remove & apply repeatable which cause of damage loss.

If the other mage have Critical Mess he will get the 5% bonus, if I applied it I will get the 5% bonus.
It's not like Mangle vs Hemorrhage which doesn't matter who apply it, both feral & rogue will benefit from it.

Atleast that what I feel don't know. Critical Mess is damn bugged.

**Comment by sigmundfreud2016** [comment]
@Illusie what you feel doesn't matter, it can be easily (dis)proven (with appropriate addon) that all spells from all sources get 5% crit chance bonus.

**Comment by PolpoVolante** [comment]
what illuse says is true, however i didn't find the same with warlock debuff only with another firemage critical mass.

Last week we did 2 fire mage in fl and on baleroc i did 10k less dps and the other mage 8k less so i think it is a big deal if you take in consideration that 1 mage has to change to a less powerful and efficient spec to not be kicked from the raid and that would be fine if the other spec wasn't that boring mess that is arcane.

**Comment by sigmundfreud2016** [comment]
@PolpoVolante boring or not, if it gives 30-40k dps by spamming 1 button, why not using it every raid? respec-price is capped at 50-60 g or something. Poor hunters are always focus-starved in all 3 specs (but especially MM) and can't relax for the whole fight.

**Comment by PolpoVolante** [comment]
@sigmundfreud2016 30-40k dps is nothing compared to the 50k firemage can pull, and its still a bug it has to be fixed. 
Arcane is weak and I don't think anyone enjoys playing a class using only 1 button. 

Who cares about hunters? its a mage thread.

**Comment by sigmundfreud2016** [comment]
@PolpoVolante 
1) 50k for 10 seconds every 2 minutes (and you have to be very-very lucky with Ignite stacks just before CD expires) and 15-20k for the other 110 seconds
2) 30-40k constantly by pressing 1 button (2 buttons if you count evocation once every 2 min)
I think the second choice is a no-brainer. PVE is not so funny anyway, it's just endless grind. Play the most effective spec to make it faster for you and your party/raid mates.

**Comment by PolpoVolante** [comment]
@sigmundfreud2016 I don't think you play on atlantiss at all or you have any insight on mage in general this expansion.
This is a bug thread so lets just stop it.

If you still think mage does 50k for 10 seconds then i can send you some screenshots and link in private of actual gameplay of Atlantiss server.

**Comment by sigmundfreud2016** [comment]
@PolpoVolante 
I may have underestimated damage of both specs in raids, since I don't like raiding. Mb there are some insane buffs/debuffs which increase dmg of players in many times, so you can scale both numbers if you want. But I am sure that arcane is by far №1 vs single target. It was always like this. Arcane blast hits like 3 pyroblasts. And it doesn't require any procs (except 4 stacks of its own debuff) to be casted in less than 1 GCD, so it's basically instant.

**Comment by Illusie** [comment]
@PolpoVolante hey polp...
wanna open new issue about stacking Critical Mass & Warlock 5% Crit (Flame & Shadow Talent)
would be nice.
Im kinda lazy >.<'

@sigmundfreud2016 sigmund, its bugreporter here, not a 'general discussion of mages in cataclysm' thank you.

**Comment by Mridusaikia** [comment]
Pretty sure that there is no existing bug with crit mass on Atlantiss - at least not of the nature that you have described. Crit mass is supposed to be re-applied and as you already know its not a personal buff but a debuff on the target which affects everyone in group. A second mage/warlock re-writing/refreshing your existing crit buff does not affect you in any way.

P.S.I play fire mage with 4 other fire mages in my group.

**Comment by Illusie** [comment]
@Mridusaikia it should work like Mangle and Hemorrhage, no matter who apply it
everyone will benefit from it,

Sadly since I can't really confirm anything because its just a "feel" when I play with 2nd Mage / Warlock in my group my DPS dropping, It doesn't matter who re-apply it, I 'feel' it doesn't benefit everyone, only the one who applied it.

Again on the 3rd time ill say its only a ' feel '.
